### PR TITLE
Fix pretraining more than once Issue #538

### LINKED ISSUE #538: [Bug] Unable to pretrain a model with dataset twice
**Describe the bug**

ValueError: Variable pretrain/model/pi_fc0/w/Adam/ already exists , disallowed. Did you mean to set reuse=True or reuse=tf.AUTO_REUSE in VarScope? Originally defined at:
File "/usr/local/lib/python3.6/dist-packages/stable_baselines/common/base_class.py", line 266, in pretrain
    optim_op = optimizer.minimize(loss, var_list=self.params)

**Code example**

```
model.pretrain(dataset, n_epochs=10)
# some code that changes values in the dataset
model.pretrain(dataset, n_epochs=10)
```

### PR DESCRIPTION
**Pull request by imontesino**
## Description

Changed the `reuse` parameter when creating the tf variables in the pretrain method to `tf.AUTO_REUSE`. This prevents tf from trying to create the variables again when pretraining more than once and raising the error:
```bash
ValueError: Variable pretrain/model/shared_fc0/w/Adam/ already exists, disallowed. Did you mean to set reuse=True or reuse=tf.AUTO_REUSE in VarScope?
```

## Motivation and Context

Closes #538

- [x] I have raised an issue to propose this change ([required](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the [changelog](https://github.com/hill-a/stable-baselines/blob/master/docs/misc/changelog.rst) accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have ensured `pytest` and `pytype` both pass (by running  `make pytest` and `make type`).

### CODE REVIEW

File: docs/misc/changelog.rst [hunk 1]
```diff
@@ -5,6 +5,25 @@ Changelog
 
 For download links, please look at `Github release page <https://github.com/hill-a/stable-baselines/releases>`_.
 
+Pre-Release 2.10.3a0 (WIP)
```
Comment (araffin) [inline docs/misc/changelog.rst:8]: please update the version in the `version.txt` file too ;)

**Review by araffin — COMMENTED** [review]
thanks for the PR =)
could you add test for that? and then it should be ready to merge ;)

File: docs/misc/changelog.rst (hunk 1 above)
Comment (imontesino) [reply docs/misc/changelog.rst:8]: Done!

**Comment by araffin** [comment]
the build is failing due to pytype, you can probably add an exception for it (and check locally that the type check passes)

**Comment by imontesino** [comment]
The pytype error seems to be in `record_expert.py`. The code in this PR passes locally as well as in CI (see the line 55 of the typecheck CI) 
```
[39/107] check stable_baselines.common.base_class
```

**Comment by araffin** [comment]
>The pytype error seems to be in record_expert.py. The code in this PR passes locally as well as in CI (see the line 55 of the typecheck CI)

yes, I know, the error is due to a newer pytype version I assume, so what I was suggesting is ignoring explicitly that error (cf pytype doc, you need to add `# pytype: disable=...`

**Comment by araffin** [comment]
The build is failing because of atari-py :see_no_evil:  (see https://github.com/openai/atari-py/pull/79)
Could you pin the atari-py version as in https://github.com/DLR-RM/stable-baselines3/pull/448 ? (`atari-py==0.2.5`)

Sorry for the CI issues...

**Review by araffin — APPROVED** [review]
LGTM, thanks =)

**Pull request merged**